clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -109,9 +109,9 @@ public Metrics(MetricConfig defaultConfig, List<MetricsReporter> reporters, Time
      */
     public Metrics(MetricConfig defaultConfig, List<MetricsReporter> reporters, Time time, boolean enableExpiration) {
         this.config = defaultConfig;
-        this.sensors = new CopyOnWriteMap<>();
-        this.metrics = new CopyOnWriteMap<>();
-        this.childrenSensors = new CopyOnWriteMap<>();
+        this.sensors = new ConcurrentHashMap<>();
```

Comment:
This should work fine but can you share some numbers with the simple stress test that we wrote? Did you consider any other approaches? I’m a bit wary of ConcurrentHashMap not offering any hard guarantees on cost of inserts/gets but in practice I think it performs well (certainly orders of magnitude better than a c-o-w map)